storage/innobase/log/log0online.c

```diff
@@ -98,7 +98,13 @@ static const char* bmp_file_name_stem = "ib_modified_log_";
 /** File name template for bitmap files.  The 1st format tag is a directory
 name, the 2nd tag is the stem, the 3rd tag is a file sequence number, the 4th
 tag is the start LSN for the file. */
+// There is simply no good way to handle truncation for this - if it happens,
+// it will break at more places than just this regex, and I don't think we
+// could handle it any better way than the default truncation
+#pragma GCC diagnostic push
```

Comment:
Not sure I understand the reasoning why we need this. But general comment - GCC is not the only compiler we support. Moreover, it doesn't look like MySQL code uses pragmas to suppress warnings

Reply:
The reason is in the place where that format string is used.

GCC added a new warning, which tries to find possible `sprintf` overflows - and this is great, I found the `sprintf` changes you can see here thanks to that.

Its report is correct even in this case - we just can't do anything about it, and it's unlikely to happen.

This is the format string:

```
"%s%s%lu_%llu.xdb"
```

It's used with `snprintf` in a situation where:

* the output buffer is 512 byte long
* and the first parameter (for the first `%s`) is also a 512 byte long buffer

Based on these, GCC deduces that it's possible that the remaining parameters will have no space in the buffer, and since we correctly pass the same 512 as the length value for `snprintf`, the output will be truncated.

* I can't really change the size of the buffers - it isn't just 512, it uses a preprocessor definition for the maximum file name length. So it's logical for both of the buffers to use it, they are files
* Because it's also based on this definition, I can't calculate the maximum space which we could allocate for the first parameter, to truncate it in a different way, as I can't do arithmetic using the preprocessor to include the result of it inside a format string. Also, this would truncate it differently, but wouldn't remove the issue of the truncation.
* Disabling this warning globally probably isn't a good idea - it's useful in other cases.

So that's why I decided to disable it locally.

That leaves me with two options:

* disable it using pragmas. While these look like gcc specific, they will also work with clang, which is the only other compiler implementing this check
* disable it using some hack that will prevent GCC's analyzer picking up this change - for example by casting one of the parameters to a pointer instead of an array, so GCC wouldn't be able to guess it's size, and would ignore this statement.

As I considered the second option a hack, and I wanted to go with a solution that disables the warning, but does so explicitly, I went with the first option.

But if you see other options, I'm open to suggestion, I didn't really like this approach, I just couldn't find anything better.

Reply:
The bmp_file_home buffer can be changed to 2 * FN_REFLEN + a bit, then add an ut_ad after the snprintf that the result length is < FN_REFLEN

Reply:
The bmp_file_home is the parameter which is too long, not the target buffer. If we want to change something, we should change log_online_bitmap_file_t.name, which I'm not sure if we should - it used at several other places, so it would require more investigation to see if it has any consequences.

Reply:
Yes, it's the target buffer I had in mind. This is our code, not upstream, larger changes are OK here